server/src/test/java/org/opensearch/gateway/ReplicaShardBatchAllocatorTests.java

```diff
@@ -692,6 +745,87 @@ private RoutingAllocation onePrimaryOnNode1And1Replica(AllocationDeciders decide
         );
     }
 
+    private RoutingAllocation routingAllocationWithMultipleShards(AllocationDeciders deciders, int primaries, int replicas)
+        throws InterruptedException {
+        return routingAllocationWithMultipleShards(deciders, Settings.EMPTY, primaries, replicas, UnassignedInfo.Reason.CLUSTER_RECOVERED);
+    }
+
+    private RoutingAllocation routingAllocationWithMultipleShards(
+        AllocationDeciders deciders,
+        Settings settings,
+        int primaries,
+        int replicas,
+        UnassignedInfo.Reason reason
+    ) throws InterruptedException {
+        Map<ShardId, ShardRouting> shardIdShardRoutingMap = new HashMap<>();
+        Index index = null;
+        for (ShardId shardIdFromBatch : shardsInBatch) {
+            shardIdShardRoutingMap.put(
+                shardIdFromBatch,
+                TestShardRouting.newShardRouting(shardIdFromBatch, node1.getId(), true, ShardRoutingState.STARTED)
+            );
+            if (index == null) {
+                index = shardIdFromBatch.getIndex();
+            }
+        }
+        IndexMetadata.Builder indexMetadata = IndexMetadata.builder(index.getName())
+            .settings(settings(Version.CURRENT).put(settings))
+            .numberOfShards(primaries)
+            .numberOfReplicas(replicas);
+        for (ShardId shardIdFromBatch : shardsInBatch) {
+            indexMetadata.putInSyncAllocationIds(
+                shardIdFromBatch.id(),
+                Sets.newHashSet(shardIdShardRoutingMap.get(shardIdFromBatch).allocationId().getId())
+            );
+        }
+        Metadata metadata = Metadata.builder().put(indexMetadata).build();
+        // mark shard as delayed if reason is NODE_LEFT
+        boolean delayed = reason == UnassignedInfo.Reason.NODE_LEFT
+            && UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.get(settings).nanos() > 0;
+        int failedAllocations = reason == UnassignedInfo.Reason.ALLOCATION_FAILED ? 1 : 0;
+        IndexRoutingTable.Builder indexRoutingTableBuilder = IndexRoutingTable.builder(index);
+        for (ShardId shardIdFromBatch : shardsInBatch) {
```

Comment:
Are we marking all the replica shards as delayed ? Should we have mix of UnassignedShards (with & w/o delay) ?

Reply:
I will have to evaluate it how this property effects the shard assignment for the test cases we have.